tests/e2e-cypress/support/commands.js

```diff
@@ -339,3 +339,22 @@ Cypress.Commands.add('addProfileJsonFixture', (mockupData) => {
         }).as('profile')
     }
 })
+
+function cesiumTilesLoaded(viewer) {
+    return new Cypress.Promise((resolve) => {
+        viewer.scene.postRender.addEventListener(() => {
+            if (viewer.scene.globe.tilesLoaded) {
+                resolve()
+            }
+        })
+    })
+}
+
+Cypress.Commands.add('waitUntilCesiumTilesLoaded', () => {
+    cy.readWindowValue('cesiumViewer').then((viewer) => {
+        cy.wrap(null).then({ timeout: 50000 }, async () => {
+            await cesiumTilesLoaded(viewer)
+            return viewer
+        })
```

Comment:
what is achieved by wrapping a `null`?
if your goal was for Cypress to wait for an arbitrary duration, you can use `cy.wait(...)` which is synchronous

instead of using a Promise based wait, you could also ressort to a request intercept. For instance if you remove what you changed in `src/modules/map/components/cesium/CesiumMap.vue` at line 71, you could intercept requests to terrain tiles and serve a mockup (or no tiles) in the intercept. This would look something like :
```
cy.intercept('**.terrain**, { /* Mock up response, or path to a fixture */}).as('loadTerrainTile')
```
and then use something like
```
cy.goToMapView('en', { showIn3d: true })
cy.wait('@loadTerrainTile')
```
I don't know if this would simplify your workflow, it's usually a bad practice in Cypress to wait for arbitrary length of time.

Reply:
I've found this `cy.wrap(null)` on a cypress blog somewhere (I will found it again and add a comment).
As for the `timeout` it's not a fixed amount to wait but the maximum duration to wait before failing, the default  10000ms if not enough